pkg/secrets/provide_conjur_secrets_test.go

```diff
@@ -315,7 +315,11 @@ func TestRunSecretsProvider(t *testing.T) {
 
 			// Check results
 			if err == nil {
-				assert.Equal(t, tc.expectedCount, tc.provider.count())
+				diff := tc.expectedCount - tc.provider.count()
```

Comment:
I don't think we should be allowing this fuzzy matching for most test cases, especially the ones that are trying to distinguish between 0 and 1. It may be better to just not check the count on the cases where it's inconsistent.

Reply:
This could just be applied for the cases with higher counts. Would that work?

Reply:
Possibly but that feels like a hacky solution to me. If we don't have a reliable count I don't think we should be testing it if there's a reasonable way to exclude the test when an "expected count" isn't provided.

Reply:
Feel free to get a second opinion though and have someone else do the actual review here. I will not be offended. :)

Reply:
The problem is we are testing the Go timer and the inconsistencies are not run to run but in different platforms. So Jenkins runs behave a little different that running on a laptop. So when we are expecting to see seven timer ticks, occasionally we see six or eight depending on the platform.
So we're not really testing the accuracy of the Go timer but just making sure something doesn't get stuck, giving a very low or very high count.

Reply:
another option is to increase the duration times so any small variations are relatively insignificant.